### ISSUE TITLE
Bug: wrong span start for decorated AssignmentPattern parameter

### ISSUE THREAD

**Issue by ArnaudBarre** [typescript-eslint/typescript-eslint#10937]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3Abug) and found none that matched my issue.
- [x] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Relevant Package

parser

### Playground Link

https://typescript-eslint.io/play#ts=5.8.2&showAST=es&fileType=.tsx&code=GYVwdgxgLglg9mABAEwKYTgJwIZSwCgEpEBvRAXwCgIAbbAZ3sQEFTLFFg458ABNDDjyZE2ADSJ%2B6LLiyIA2gCMAuhKmDZIshApqBM4SkQBeRAEZiZKlSA&eslintrc=N4KABGBEBOCuA2BTAzpAXGYBfEWg&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA&tokens=false

### Repro Code

```TypeScript
function decorator() { }
class A {
  foo(@decorator a, @decorator [b], @decorator { c }, @decorator d = 1) { }
}
```

### Expected Result

Span start for decorated AssignmentPattern doesn't include the decorator to match other binding pattern.

### Actual Result

Span start include the decorator

### Versions

`@typescript-eslint/parser`: `8.26.1`

**Comment by bradzacher** [comment]
part of me thinks we should just "hand wave" this away and ignore it because function parameter decorators are not valid with "modern" decorators and are only supported with legacy decorators.
